sui/manage-coins.js

```diff
@@ -0,0 +1,218 @@
+        const response = tx.splitCoins(firstObjectId, [splitAmount]);
+        console.log(`Split coins of type '${chalk.green(coins.data[0].coinType)}' with amount ${splitAmount}`);
+        return response;
+    }
+
+    static async mergeCoin(tx, coinTypeToCoins, options) {
+        const coinType = options.coinType;
+        console.log('\n==== Merging Coins ====');
+
+        // Throw an error if the coin type is specified but no coins are found
+        CoinManager.checkCoinType(coinType, coinTypeToCoins);
+
+        if (coinType) {
+            const coins = coinTypeToCoins[coinType];
+            await CoinManager.doMergeCoin(tx, coins);
+        } else {
+            for (const coinType in coinTypeToCoins) {
+                const coins = coinTypeToCoins[coinType];
+                await CoinManager.doMergeCoin(tx, coins);
+            }
+        }
+    }
+
+    static async doMergeCoin(tx, coins) {
+        const coinObjectIds = coins.data.map((coin) => coin.coinObjectId);
+
+        // If the first coin is a gas token, remove it from the list. Otherwise, the merge will fail.
+        if (CoinManager.isGasToken(coins.data[0])) {
+            coinObjectIds.shift();
+        }
+
+        if (coinObjectIds.length < 2) {
+            // Need at least 2 coins to merge
+            return;
+        }
+
+        const firstCoin = coinObjectIds.shift();
+        const remainingCoins = coinObjectIds.map((id) => tx.object(id));
+
+        tx.mergeCoins(firstCoin, remainingCoins);
+        console.log(`Merge ${coins.data.length} coins of type '${chalk.green(coins.data[0].coinType)}'`);
+    }
+
+    static async processCommand(config, chain, options) {
+        const [keypair, client] = getWallet(chain, options);
+        printInfo('Wallet Address', keypair.toSuiAddress());
+
+        const coinTypeToCoins = await CoinManager.getAllCoins(client, keypair.toSuiAddress());
+        CoinManager.printAllCoins(coinTypeToCoins);
+
+        const tx = new TransactionBlock();
+
+        if (options.merge) {
+            await CoinManager.mergeCoin(tx, coinTypeToCoins, options);
+        } else if (options.split) {
+            await CoinManager.splitCoins(tx, coinTypeToCoins, options);
+        }
+
+        const requireBroadcast = options.merge || options.split;
+
+        if (requireBroadcast) {
+            await client.signAndExecuteTransactionBlock({
+                transactionBlock: tx,
+                signer: keypair,
+                options: {
+                    showEffects: true,
+                    showObjectChanges: true,
+                    showContent: true,
+                },
+            });
+
+            console.log(`\nDone`);
+        }
+    }
+
+    static printAllCoins(coinTypeToCoins) {
+        console.log('==== Coins Info ====');
+
+        for (const coinType in coinTypeToCoins) {
+            const coins = coinTypeToCoins[coinType];
+            console.log(`Coin Type: ${chalk.green(coinType)}`);
```

Comment:
`printInfo` handles colouring

Reply:
I'd like to group these three lines for each coin close together like this.

```
==== Coins Info ====
Coin Type: 0x2::sui::SUI
Total Balance: 1000000000000
Total Objects: 5
````

But the `printInfo` function adds an extra line like this.

```
==== Coins Info ====
Coin Type:: 0x2::sui::SUI

Total Balance:: 1000000000000

Total Objects:: 5
```

I guess you prefer the latter? I'll update to printInfo.